### PR TITLE
Add support to report multiple errors

### LINKED ISSUE #1: [Feature] Pass multiple error messages
There is no way currently to pass multiple errors encountered when a runner created from `create-jest-runner` is executed.

Update: The test results support multiple error messages, but the abstracted `fail` function doesn't supports passing multiple test results. So the `fail` function needs to be updated to pass all test errors.

### PR DESCRIPTION
**Pull request by SaurabhAgarwala**
Fixes #1 

Merge only after https://github.com/MLH-Fellowship/jest-runner-tsd/issues/4 is fixed.

The `fail` function has been updated to report of multiple errors in `fail.js` Correspondingly, the runner has been updated to call this custom implementation of `fail` in `run.js`. Finally, `toMultipleTestResults` has been derived from `toTestResults` to support this custom implementation of `fail` and some of the optional parameters in `toTestResults` removed.

Below is the screenshot demonstrating multiple error reporting 
image

TODO: The `pass` function in `run.js` has to be updated to report the number of tests passed. This can be done only after the required changes are made in the [tsd repo](https://github.com/MLH-Fellowship/tsd)

### CODE REVIEW

File: src/fail.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// @ts-check
+
+const { toMultipleTestResults } = require("./toMultipleTestResults");
+
+module.exports.fail = ({ start, end, failures }) =>
```
Comment (smartclash) [inline src/fail.js:5]: We can probably use brackets, makes it easier to read :)

File: src/run.js [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
-    testFiles: [ testFile ]
+    testFiles: [testFile],
   });
 
-  // Currently only reports a single error.
   if (diagnose.length > 0) {
+    const failures = diagnose.map((test) => ({
+      path: test.fileName,
+      errorMessage: test.message,
+      title: test.severity + testFile,
```
Comment (smartclash) [inline src/run.js:20]: We should add a space here.
```suggestion
      title: test.severity + ' ' + testFile,
```

Comment (SaurabhAgarwala) [reply src/run.js]: Yes, I have updated this to report more meaningful titles. It's better now

File: src/fail.js (hunk 1 above)
Comment (SaurabhAgarwala) [reply src/fail.js:5]: Not very sure of this means. Can you please suggest the change

**Comment by smartclash** [comment]
**IDEA**
We can probably use `chalk` library to make the severity text bold so that it's easy to glance through thousands of errors.

**Comment by SaurabhAgarwala** [comment]
What I have observed, is that the value of `severity` is always `error` only. Never have I got any other severity value, so I removed it. If you think it's better to have then we can.

**Comment by smartclash** [comment]
> What I have observed, is that the value of `severity` is always `error` only. Never have I got any other severity value, so I removed it. If you think it's better to have then we can.

No @SaurabhAgarwala, don't remove that. Yes it returns `error` always. But the TSD library may once mature and add more severity and we don't want to miss out on that.

File: src/fail.js (hunk 1 above)
Comment (smartclash) [reply src/fail.js:5]: Ehm, I don't know how to lay down that in words. I will make the change myself in this PR. Don't worry.

**Comment by SaurabhAgarwala** [comment]
Corrected a typo in a commit message. Hence had to force-push

**Comment by SaurabhAgarwala** [comment]
> > What I have observed, is that the value of `severity` is always `error` only. Never have I got any other severity value, so I removed it. If you think it's better to have then we can.
> 
> No @SaurabhAgarwala, don't remove that. Yes it returns `error` always. But the TSD library may once mature and add more severity and we don't want to miss out on that.

Ok, so shall we just show it in the beginning of the title message itself?

**Comment by smartclash** [comment]
Yeah, something like `[error] at /__tests__/add.test.ts` seems like a good way to show.

**Comment by smartclash** [comment]
> Yeah, something like `[error] at /__tests__/add.test.ts` seems like a good way to show.

We should then also fix #5

**Comment by SaurabhAgarwala** [comment]
Cool, will look into it

File: src/fail.js (hunk 1 above)
Comment (SaurabhAgarwala) [reply src/fail.js:5]: Ok

File: src/run.js [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
-    testFiles: [ testFile ]
+    testFiles: [testFile],
   });
 
-  // Currently only reports a single error.
   if (diagnose.length > 0) {
+    const failures = diagnose.map((test) => ({
+      path: test.fileName,
+      errorMessage: test.message,
+      title: `${testFile}:${test.line}:${test.column} - ${test.severity} - ${test.message}`
```
Comment (smartclash) [inline src/run.js:20]: @SaurabhAgarwala let me know if this seems like a good title for you

Comment (SaurabhAgarwala) [reply src/run.js:20]: Yeah this looks good

**Comment by smartclash** [comment]
Just merging it. Been stalling for a long time. Will fix stuffs along the way.

**Pull request merged**